Support for i.MX 8M chipsets

passgat opened the pull request:
The series adds bootcount support for the i.MX8M platform (last patch). 
It also includes rework patches to facilitate the addition of a new platform and two patches to handle the 
endianness of the bootcount value. 
I have tested the series on the STM32MP1, am33xx (BeagleBone), and i.MX8M platforms.

passgat commented:
Gentle ping.

Thanks,
Dario

thom-nic commented:
This looks really great, I appreciate the contribution.  I'll give a quick manual test on our end and assuming it all works I'll be happy to merge it.

thom-nic commented:
@passgat could you provide:

* reference links (if available online) to the technical manual docs for the i.MX8 registers being used
* corresponding u-boot config settings?

I realized I am missing the same for am33 and mp157 but I intend to add that... I have it here somewhere...

passgat commented:
@thom-nic
For i.MX8m:
I added documentation, list of reference manuals, in the file header of imx8m.c.
I added U-Boot configurations (the bootcount address) in the commit message

passgat commented:
A gentle ping.

Thanks,
Dario

thom-nic commented:
@passgat thanks for the reminder. I will try to fit this in soon.

Your contribution looks great of course.  I don't have much in the way of feedback w/r/t the i.MX 8 code specifically since I don't have that hardware. I mostly need to build your branch and regression test it for our hw platforms that we use.

passgat commented:
I added the imx8m platform because I needed the bootcounnt support for it. So I run tests on this platform. I run tests on the beaglebome (am33xx) and the stm32mp157f-dk2 too, so I don't expect regressions. I don't remember If I run test on eeprom, but I don't think so. 

Thanks,
Dario

thom-nic commented:
That's fine, thank you.  I'll do some quick testing on my end so I can merge the PR

thom-nic reviewed (changes requested):
@passgat would you mind replacing the \t characters with spaces?  I see it in am33xx.c and bootcount.c

Thank you!

passgat commented:
Done,
I hope I haven't forgotten any tabs.

thom-nic commented:
Finally got to testing this on an EEPROM device.  Looks great.  Thanks for the contribution!

thom-nic commented:
@passgat I will do a release. I also really should update the README for IMX8 and STM32MP1 as well..

thom-nic commented:
@passgat release created: https://github.com/VoltServer/uboot-bootcount/releases/tag/v3.0.0

passgat commented:
@thom-nic Many thanks!
Best regards
